### PR TITLE
NN-3773: MVP+1 incident details updates

### PR DESCRIPTION
**Pull request by natclamp-moj**

### CODE REVIEW

File: server/routes/incidentDetails/incidentDetails.ts [hunk 1]
```diff
@@ -42,40 +72,101 @@ export default class IncidentDetailsRoutes {
...
+          assistAnotherPrisonerInput,
+        })
+      req.session.redirectUrl = `/incident-details/${prisonerNumber}?queryRadioSelection=${currentRadioSelected}`
+      req.session.incidentDate = incidentDate
+      req.session.incidentLocation = locationId
+      const searchPageHref = `/select-associated-prisoner/${prisonerNumber}?searchTerm=${searchValue}`
+      return res.redirect(`${searchPageHref}&startUrl=incident-details`)
+    }
+
+    // Check the selectedPerson has not been tampered with in URL
```
Comment (richardpopple) [inline server/routes/incidentDetails/incidentDetails.ts:137]: Interested to know what is going on here - is this in case the user retypes when back on the radio button page?

File: server/routes/incidentDetails/incidentDetailsSearchValidation.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  },
+  MISSING_PRISONER_ASSISTED: {
+    href: '#assistAnotherPrisonerInput',
+    text: 'Enter a prisoner’s name or number',
+  },
+}
+
+export default function validateForm({ searchTerm, inputId }: SearchForm): FormError | null {
+  if (!searchTerm) {
+    return inputId === 'inciteAnotherPrisoner' ? errors.MISSING_PRISONER_INCITED : errors.MISSING_PRISONER_ASSISTED
```
[the remark this answers is not in the document]
Comment (sp-watson) [reply server/routes/incidentDetails/incidentDetailsSearchValidation.ts:21]: ..possibly more trouble then it's worth

File: server/routes/incidentDetails/incidentDetails.ts [hunk 3]
```diff
@@ -42,40 +49,92 @@ export default class IncidentDetailsRoutes {
...
     try {
       const newAdjudication = await this.placeOnReportService.startNewDraftAdjudication(
         formatDate(incidentDate),
         locationId,
         prisonerNumber,
+        associatedPrisonersNumber,
+        currentRadioSelected, // Somewhere before(?) this we need to convert the currentRadioSelected to the calculated roleCode
         user
       )
+      delete req.session.redirectUrl
```
Comment (sp-watson) [inline server/routes/incidentDetails/incidentDetails.ts:137]: Delete as part of the get(), which is where it should be read in answering another PR

**Review by sp-watson — APPROVED** [review]
